front/src/applications/operationalStudies/views/Scenario/components/Timetable/PacedTrain/PacedTrainItem.tsx

```diff
@@ -153,6 +159,7 @@ const PacedTrainItem = ({
     try {
       await deletePacedTrains(dispatch, [pacedTrain.id]);
       removePacedTrains([pacedTrain.id]);
+      setSelectedTimetableItemIds(selectedTimetableItemIds.filter((id) => id !== pacedTrain.id));
```

Comment:
If we do it like below, we can avoid passing down `selectedTimetableItemIds` in this component and `TrainScheduleItem`.
```suggestion
      setSelectedTimetableItemIds((prev) => prev.filter((id) => id !== pacedTrain.id));
```

To be able to do that, we need to use the proper type of "setState" which here would be `React.Dispatch<SetStateAction<TimetableItemId[]>>`

Reply:
fixed !